contracts/scripts/app.ts

```diff
@@ -8,92 +8,132 @@
-// --- Constants ---
-// Define grants to create (addresses are random)
-const grants = [
-  {
-    owner: '0x34f4E532a33EB545941e914B25Efe348Aea31f0A',
-    payee: '0x06c94663E5884BE4cCe85F0869e95C7712d34803',
-    metaPtr: 'https://invent-teleportation.eth.link',
-  },
-  {
-    owner: '0x58E52440F56f2A5307772Ec881BCEf2c15e988Ab',
-    payee: '0x6f02c37ea174DD05f20aC118da725ffa6A40B990',
-    metaPtr: 'https://get-to-mars.eth.link',
-  },
-  {
-    owner: '0x1fB6C46e6aDD95698352707D7f93a31030c80a0B',
-    payee: '0x834e659c6757E250db500fe869877311Bb552966',
-    metaPtr: 'https://time-travel.eth.link',
-  },
-];
-
-const createGrantRoundFactory = async (deployer: SignerWithAddress, registry_address: string) => {
+const fixtureGrants = async (deployer: SignerWithAddress) => {
+  // Define grants to create (addresses are random)
+  const grants = [
+    {
+      owner: '0x34f4E532a33EB545941e914B25Efe348Aea31f0A',
+      payee: '0x06c94663E5884BE4cCe85F0869e95C7712d34803',
+      metaPtr: 'https://invent-teleportation.eth.link',
+    },
+    {
+      owner: '0x58E52440F56f2A5307772Ec881BCEf2c15e988Ab',
+      payee: '0x6f02c37ea174DD05f20aC118da725ffa6A40B990',
+      metaPtr: 'https://get-to-mars.eth.link',
+    },
+    {
+      owner: '0x1fB6C46e6aDD95698352707D7f93a31030c80a0B',
+      payee: '0x834e659c6757E250db500fe869877311Bb552966',
+      metaPtr: 'https://time-travel.eth.link',
+    },
+  ];
+
+  // Deploy contract
+  const GrantRegistryFactory: ContractFactory = await ethers.getContractFactory('GrantRegistry', deployer);
+  const registry = await (await GrantRegistryFactory.deploy()).deployed();
+  console.log(`Deployed GrantRegistry to ${registry.address}`);
+
+  // Create the grants
+  await Promise.all(grants.map((grant) => registry.createGrant(grant.owner, grant.payee, grant.metaPtr)));
+  console.log(`Created ${grants.length} dummy grants`);
+
+  return registry;
+};
+
+const fixtureManager = async (deployer: SignerWithAddress, registry: Contract) => {
   // --- SwapRouter --
   const router = await ethers.getContractAt(SWAP_ROUTER_ABI, UNISWAP_ROUTER, deployer);
   console.log(`Fetched SwapRouter at ${router.address}`);
 
   // --- GrantRoundManager --
   const GrantRoundManager: ContractFactory = await ethers.getContractFactory('GrantRoundManager', deployer);
   const roundManager = await (
-    await GrantRoundManager.deploy(registry_address, router.address, tokens.gtc.address)
+    await GrantRoundManager.deploy(registry.address, router.address, tokens.dai.address)
   ).deployed();
+
   console.log(`Deployed GrantRoundManager to ${roundManager.address}`);
+  return roundManager;
+};
 
+const fixtureRound = async (registry: Contract, manager: Contract) => {
   // --- GRANT ROUND ---
-  const startDate = new Date();
-  const endDate = new Date();
-  endDate.setMonth(endDate.getMonth() + 2);
+  let startTime = Math.floor(new Date().getTime() / 1000); // time in seconds
+  startTime = await setNextBlockTimestamp(startTime + 200);
```

Comment:
Something to keep in mind as we move forward: fast forwarding time like this effects the local blockchain, but (obviously) not the system time the app will get when reading `Date().getTime()`. So if other parts of the app pull the time and then try to interact with something on chain, this could cause issues (for example, attempting to deploy a GrantRound with a start time that is in the "past" according to the chain). Open to suggestions on how to best manage this. Just wanted to call it out as a potential issue moving forward.

Reply:
Here we are setting the startTime to be a time in the future, so there is a delay when starting GrantRound, correct? seems issue might be interacting with a GrantRound that hasn't started yet? We could I guess try adjusting the system time to keep it consistent (create a separate getter, and recommend devs to use this), perhaps not the best solution but just a thought.